src/main/shortcutHandler.js

```diff
@@ -0,0 +1,295 @@
+
+      // format menu
+      ['formatStrong', 'CmdOrCtrl+B'],
+      ['formatEmphasis', 'CmdOrCtrl+I'],
+      ['formatInlineCode', 'CmdOrCtrl+`'],
+      ['formatStrike', 'CmdOrCtrl+D'],
+      ['formatHyperlink', 'CmdOrCtrl+L'],
+      ['formatImage', 'CmdOrCtrl+Shift+I'],
+      ['formatClearFormat', 'Shift+CmdOrCtrl+R'],
+
+      // window menu
+      ['windowMinimize', 'CmdOrCtrl+M'],
+      ['windowCloseWindow', 'CmdOrCtrl+Shift+W'],
+
+      // view menu
+      ['viewToggleFullScreen', isOsx ? 'Ctrl+Command+F' : 'F11'],
+      ['viewChangeFont', 'CmdOrCtrl+.'],
+      ['viewSourceCodeMode', 'CmdOrCtrl+Alt+S'],
+      ['viewTypewriterMode', 'CmdOrCtrl+Alt+T'],
+      ['viewFocusMode', 'CmdOrCtrl+Shift+F'],
+      ['viewToggleSideBar', 'CmdOrCtrl+J'],
+      ['viewToggleTabBar', 'CmdOrCtrl+Alt+B'],
+      ['viewDevToggleDeveloperTools', isOsx ? 'Alt+Command+I' : 'Ctrl+Shift+I'],
+      ['viewDevReload', 'CmdOrCtrl+R']
+    ])
+
+    // fix non-US keyboards
+    this.mnemonics = new Map()
+    this.fixLayout()
+
+    // load user-defined keybindings
+    this.loadLocalKeybindings()
+  }
+
+  fixLayout () {
+    const lang = getKeyboardLanguage()
+    switch (lang) {
+      // fix aidou and inline code
+      case 'ch':
+      case 'de':
+      case 'dk':
+      case 'fi':
+      case 'no':
+      case 'se':
+        this.fixInlineCode()
+
+        if (!isOsx) {
+          this.fixAidou()
+        }
+        break
+
+      // fix aidou only
+      case 'es':
+      case 'hr':
+      case 'it':
+      case 'pl':
+      case 'pt':
+        if (!isOsx) {
+          this.fixAidou()
+        }
+        break
+
+      // custom layouts
+      case 'bg':
+        this.mnemonics.set('CmdOrCtrl+/', 'CmdOrCtrl+8')
+        this.fixInlineCode()
+        break
+    }
+  }
+
+  fixAidou () {
+    this.mnemonics.set('CmdOrCtrl+/', 'CmdOrCtrl+7')
+  }
+
+  // fix dead backquote key on layouts like German
+  fixInlineCode () {
+    this.keys.set('formatInlineCode', 'CmdOrCtrl+Shift+B')
+  }
+
+  loadLocalKeybindings () {
+    if (global.MARKTEXT_SAFE_MODE || !isFile(this.configPath)) {
+      return
+    }
+
+    const json = readJson(this.configPath, true)
+    if (!json || typeof json !== 'object') {
+      log('Invalid keybindings.json configuration.')
+      return
+    }
+
+    // keybindings.json example:
+    // {
+    //   "fileSave": "CmdOrCtrl+S",
+    //   "fileSaveAs": "CmdOrCtrl+Shift+S"
+    // }
+
+    const userAccelerators = new Map()
+    for (const key in json) {
+      if (this.keys.has(key)) {
+        const value = json[key]
+        if (typeof value === 'string' && isAccelerator(value)) {
+          // id / accelerator
+          userAccelerators.set(key, value)
+        }
+      }
+    }
+
+    if (userAccelerators.size === 0) {
+      return
+    }
+
+    // deep clone shortcuts and unset user shortcuts
+    const accelerators = new Map(this.keys)
+    userAccelerators.forEach((value, key) => {
+      accelerators.set(key, '')
```

Comment:
Why need to set the use set `key` to empty `string`?

Reply:
Because we have to release the accelerators before duplicate check. If we don't unset the accelerators the following example will not work:

```
{
  "fileSave": "CmdOrCtrl+Shift+S",
  "fileSaveAs": "CmdOrCtrl+S"
}
```

because `CmdOrCtrl+S` and `CmdOrCtrl+Shift+S` are already in use.

Reply:
>Because we have to release the accelerators before duplicate check.

If we have already set the accelerator to empty string, what's the meaning of duplicate check?

Reply:
```javascript
if (this.equalsAccelerators(value, userValue))  {}
```

is codes above always be `false`, because we set it to empty string?

Reply:
No, we unset the keys because we know that we set them to another accelerator. Then we check for duplicates because we may changed `fileSave` to `Ctrl+T` but `Ctrl+T` already exists as `paragraphTable`. In this example `equalsAccelerators(...)` is `true` and we have to stop. In the example above `equalsAccelerators(...)` is always `false` because there are not duplicated accelerators.

Reply:
ha, I understand it after your explanation! thanks.